### PR TITLE
Added new IAMModule and updated all integrations

### PR DESCRIPTION
**Pull request by Shellyber**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/32483

## Description
Added IAMModule
Removed all IAM classes from CommonServerPython
Updated relevant integrations.

## Minimum version of Demisto
- [x] 6.0.0

## Does it break backward compatibility?
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/GitHub/Integrations/GitHub_IAM/GitHub_IAM.py [hunk 1]
```diff
@@ -362,20 +363,20 @@ def main():
 
         elif command == 'iam-get-user':
             user_profile = get_user_command(client, args, mapper_in)
-            return_results(user_profile)
+            user_profile.return_outputs
```
Comment (dantavori) [inline Packs/GitHub/Integrations/GitHub_IAM/GitHub_IAM.py:366]:
```suggestion
            user_profile.return_outputs()
```
please fix it in the following ones as well

File: Packs/ApiModules/Scripts/IAMModule/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+To use the common IAM API logic, run the following command to import the `IAMModule`.
```
Comment (dantavori) [inline Packs/ApiModules/Scripts/IAMModule/README.md:1]: let's include here information about the classes in IAMModule (use the docstrings if you aren't sure what is the purpose of each one)

File: package-lock.json [hunk 3]
```diff
@@ -467,6 +467,67 @@
       "resolved": "https://registry.npmjs.org/character-reference-invalid/-/character-reference-invalid-1.1.4.tgz",
       "integrity": "sha512-mKKUkUbhPpQlCOfIuZkvSEgktjPFIsZKRRbC6KWVEMvlzblj3i3asQv5ODsrwt0N3pHAEvjP8KTQPHkp0+6jOg=="
     },
+    "clix": {
```
Comment (dantavori) [inline package-lock.json:470]: what are these changes?

**Review by dantavori — CHANGES_REQUESTED** [review]
looks good 👍🏽  see my notes, and let's make sure the integrations still work the same after the changes (you can create a self service and configure the marketplace to the build bucket to install them)

File: package-lock.json (hunk 3 above)
Comment (Shellyber) [reply package-lock.json:470]: Don't know, will revert it.

**Comment by Shellyber** [comment]
@dantavori Please review

File: Packs/HelloIAMWorld/pack_metadata.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
     "name": "Hello IAM World",
```
Comment (anara123) [inline Packs/HelloIAMWorld/pack_metadata.json:2]:
```suggestion
    "name": "Hello World IAM",
```

Comment (anara123) [reply Packs/HelloIAMWorld/pack_metadata.json:2]: Lets change the pack name to be aligned with other Hello World examples

**Comment by anara123** [comment]
@Shellyber what is the status here?

File: Packs/ApiModules/Scripts/IAMApiModule/README.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    ...
+
+
+from IAMApiModule import *  # noqa: E402
+
+if __name__ in ["builtins", "__main__"]:
+    main()
+```
+
+Then, the `IAMApiModule` will be available for usage. For examples, see the `Workday` or `Okta` integrations.
```
[the remark this answers is not in the document]
Comment (anara123) [reply Packs/ApiModules/Scripts/IAMApiModule/README.md:14]: Add link to those integration python files

File: Packs/Base/ReleaseNotes/1_6_8.md [hunk 6]
```diff
@@ -0,0 +1,4 @@
+
+#### Scripts
+##### CommonServerPython
+- General performance and reliability improvements.
```
Comment (anara123) [inline Packs/Base/ReleaseNotes/1_6_8.md:4]: Add proper release notes. Explain that we moved IAM classes into separate module

File: Packs/ApiModules/Scripts/IAMApiModule/IAMApiModule.py [hunk 7]
```diff
@@ -0,0 +1,471 @@
...
+        if self._vendor_action_results[0].should_return_error():
+            return_entry['Type'] = EntryType.ERROR
+        else:
+            return_entry['Type'] = EntryType.NOTE
+            return_entry['HumanReadable'] = readable_output
+
+        return return_entry
+
+    def return_outputs(self):
+        return_results(self.to_entry())
```
Comment (anara123) [inline Packs/ApiModules/Scripts/IAMApiModule/IAMApiModule.py:171]: why do even have this?

File: Packs/GitHub/Integrations/GitHub_IAM/GitHub_IAM.py [hunk 8]
```diff
@@ -362,20 +362,20 @@ def main():
 
         elif command == 'iam-get-user':
             user_profile = get_user_command(client, args, mapper_in)
-            return_results(user_profile)
+            user_profile.return_outputs()
```
Comment (anara123) [inline Packs/GitHub/Integrations/GitHub_IAM/GitHub_IAM.py:365]: revert these changes. keep using `return_results`
we should be consistent with how we return stuff from integration/scripts

**Review by anara123 — CHANGES_REQUESTED** [review]

File: Packs/ApiModules/Scripts/IAMApiModule/IAMApiModule.py (hunk 7 above)
Comment (Shellyber) [reply Packs/ApiModules/Scripts/IAMApiModule/IAMApiModule.py:171]: @anara123 Once we removed all the classes from CommonServerPython, we needed to remove also the part from `return_results` function.
Since we can import in CommonServerPython, @dantavori and I, thought that the best solution to keep it the same and return it as a proper entry to the war room is with adding this function.

File: Packs/GitHub/Integrations/GitHub_IAM/GitHub_IAM.py (hunk 8 above)
Comment (Shellyber) [reply Packs/GitHub/Integrations/GitHub_IAM/GitHub_IAM.py:365]: See my comment above.

**Comment by Shellyber** [comment]
@anara123 Please review again.
See my comment regarding the `return_results` function.
For my part the PR is ready.

File: Packs/GitHub/Integrations/GitHub_IAM/GitHub_IAM.py (hunk 8 above)
Comment (Shellyber) [reply Packs/GitHub/Integrations/GitHub_IAM/GitHub_IAM.py:365]: So we can change this to be:
`return_results(user_profile.to_entry())' if you prefer

File: Packs/ApiModules/Scripts/IAMApiModule/IAMApiModule.py (hunk 7 above)
Comment (anara123) [reply Packs/ApiModules/Scripts/IAMApiModule/IAMApiModule.py:171]: Instead, lets make the return_results less coupled.
In return_results check if the object or objects that passed has attribute `get_entry` function. If yes, then call that function. This way any object that implements `get_entry` function will work

**Comment by Shellyber** [comment]
@anara123 Please review

**Review by anara123 — APPROVED** [review]

**Comment by Itay4** [comment]
build fails due to https://github.com/demisto/etc/issues/33163
merging
